File: FarNetCSharp/Command1.cs
using FarNet;
using System;
using System.Data.Common;

namespace FarNetCSharp;

[ModuleCommand(Name = "FarNetCSharp", Prefix = "FarNetCSharp", Id = "30bc4a2c-6e28-43cb-ad38-5c0ee6b7f28f")]
public class Command1 : ModuleCommand
{
	public override void Invoke(object sender, ModuleCommandEventArgs e)
	{
		try
		{
			// The format of command line is up to the module.
			// Here is the parser of connection string format.
			var parameters = new DbConnectionStringBuilder { ConnectionString = e.Command };

			Far.Api.UI.WriteLine($"Parsed {parameters.Count} parameters:");
			foreach (string key in parameters.Keys)
				Far.Api.UI.WriteLine($"{key} = {parameters[key]}");
		}
		catch (Exception ex)
		{
			Far.Api.UI.WriteLine(
				$"Please use semicolon separated key=value pairs. Error: {ex.Message}",
				ConsoleColor.Red);
		}
	}
}

File: FarNetFSharp/Command1.fs
namespace FarNetFSharp
open FarNet
open System
open System.Data.Common

[<ModuleCommand(Name = "FarNetFSharp", Prefix = "FarNetFSharp", Id = "0d526cb9-7006-4ab4-8b82-7bc5d3cb8e2b")>]
type Command1() =
    inherit ModuleCommand()
    override _.Invoke(_, e) =
        try
            // The format of command line is up to the module.
            // Here is the parser of connection string format.
            let parameters = DbConnectionStringBuilder(ConnectionString = e.Command)

            Far.Api.UI.WriteLine($"Parsed {parameters.Count} parameters:")
            for key in parameters.Keys do
                Far.Api.UI.WriteLine($"{key} = {parameters[string key]}");

        with ex ->
            Far.Api.UI.WriteLine(
                $"Please use semicolon separated key=value pairs. Error: {ex.Message}",
                ConsoleColor.Red)
